### PR TITLE
Add Enable/Disable Keyboard Shortcuts to settings

### PR DESCRIPTION
**Pull request by Hector118**
## Pull Request Type

- [ ] Bugfix
- [x] Feature Implementation
- [ ] Documentation
- [ ] Other

## Related issue

- Closes https://github.com/FreeTubeApp/FreeTube/issues/992

## Description
Added a setting to the player settings section which toggles keyboard shortcuts on the video player.

## Desktop
- **OS:**  Ubuntu LTS Jammy Jellyfish
- **OS Version:** 22.04
- **FreeTube version:** 0.19.1

### CODE REVIEW

File: src/renderer/store/modules/settings.js [hunk 1]
```diff
@@ -226,6 +226,7 @@ const state = {
   hideLabelsSideBar: false,
   hideChapters: false,
   showDistractionFreeTitles: false,
+  keyboardShortcuts: false,
```
Comment (ChunkyProgrammer) [inline src/renderer/store/modules/settings.js:229]: Maybe this should be set to true as some users are already using keyboard shortcuts and will probably open an issue thinking that shortcuts are broken, not realizing they have to enable it

Comment (absidue) [reply src/renderer/store/modules/settings.js]: Definitely should be enabled by default, as the keyboard shortcuts have existed for a long time and most users want them.

A better option would be to name the setting `Disable Keyboard Shortcuts`, not only does that clearly indicate to the user that the default is to have the keyboard shortcuts enabled, but it would also make the code clearer as you would have `disableKeyboardShortcuts`, instead of the ambiguously named `keyboardShortcuts` value, that doesn't immediately tell you that it is a boolean.

**Review by absidue — COMMENTED** [review]
As mentioned above, could you please rename the setting to `Disable Keyboard Shortcuts` and update the code accordingly.

> A better option would be to name the setting Disable Keyboard Shortcuts, not only does that clearly indicate to the user that the default is to have the keyboard shortcuts enabled, but it would also make the code clearer as you would have disableKeyboardShortcuts, instead of the ambiguously named keyboardShortcuts value, that doesn't immediately tell you that it is a boolean.

**Comment by Hector118** [comment]
> As mentioned above, could you please rename the setting to `Disable Keyboard Shortcuts` and update the code accordingly.
> 
> > A better option would be to name the setting Disable Keyboard Shortcuts, not only does that clearly indicate to the user that the default is to have the keyboard shortcuts enabled, but it would also make the code clearer as you would have disableKeyboardShortcuts, instead of the ambiguously named keyboardShortcuts value, that doesn't immediately tell you that it is a boolean.

I have changed the name of the setting to "disable Keyboard Shortcuts" in the player settings section and in other files.

File: src/renderer/components/ft-video-player/ft-video-player.js [hunk 2]
```diff
@@ -2087,8 +2090,8 @@ export default defineComponent({
       if ((event.ctrlKey || event.metaKey) && event.key.toLowerCase() === 'c') {
         return
       }
-
-      if (this.player !== null) {
+      // 'this.disableKeyboardShortcuts === true' address disabling keyboard shortcuts
+      if (this.player !== null && this.disableKeyboardShortcuts === true) {
```
Comment (ChunkyProgrammer) [inline src/renderer/components/ft-video-player/ft-video-player.js:2094]:
```suggestion
      if (this.player !== null && this.disableKeyboardShortcuts === false) {
```

File: src/renderer/store/modules/settings.js [hunk 3]
```diff
@@ -226,6 +226,7 @@ const state = {
   hideLabelsSideBar: false,
   hideChapters: false,
   showDistractionFreeTitles: false,
+  disableKeyboardShortcuts: true,
```
Comment (ChunkyProgrammer) [inline src/renderer/store/modules/settings.js:229]:
```suggestion
  disableKeyboardShortcuts: false,
```

File: src/renderer/components/ft-video-player/ft-video-player.js [hunk 4]
```diff
@@ -2087,8 +2090,8 @@ export default defineComponent({
       if ((event.ctrlKey || event.metaKey) && event.key.toLowerCase() === 'c') {
         return
       }
-
-      if (this.player !== null) {
+      // 'this.disableKeyboardShortcuts === true' address disabling keyboard shortcuts
```
Comment (ChunkyProgrammer) [inline src/renderer/components/ft-video-player/ft-video-player.js:2093]:
```suggestion
      // 'this.disableKeyboardShortcuts === false' address disabling keyboard shortcuts
```

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — COMMENTED** [review]
Hmm few shortcuts dont seem to be disabled. Should we even disable them?
 
- Focus searchbar `alt + d`
- Refresh `r`

Most of the bottom table shouldn't be touched because those are necessary but maybe these can also go 

- Command/Ctrl + , 	Navigate to Preferences
- Command + Y (MacOS)/Ctrl + H (Windows & Linux) 	Navigate to History
- Alt + Left Arrow 	Go back one page
- Alt + Right Arrow 	Go forward one page

All shortcuts can be found https://docs.freetubeapp.io/usage/keyboard-shortcuts/

**Comment by absidue** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc This pull request currently only targets the player ones and puts the setting in the player settings.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Do we consider that enough for that issue to close?

**Comment by jasonhenriquez** [comment]
I don't think we can justify the other interpretation as a feature we would want to implement, so that's fine in my book.

**Comment by Hector118** [comment]
Made the changes that were suggested for disableKeyboardShortcuts.

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — COMMENTED** [review]
Could u move the toggle to the bottom of the list.

Capture

**Comment by Hector118** [comment]
> Could u move the toggle to the bottom of the list.
> 
> Capture

moved it to the bottom

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

File: src/renderer/components/ft-video-player/ft-video-player.js [hunk 5]
```diff
@@ -343,6 +343,10 @@ export default defineComponent({
 
     proxyVideos: function () {
       return this.$store.getters.getProxyVideos
+    },
+
+    disableKeyboardShortcuts: function() {
+      return this.$store.getters.getDisableKeyboardShortcuts
     }
```
Comment (PikachuEXE) [inline src/renderer/components/ft-video-player/ft-video-player.js:350]:
```suggestion
    keyboardShortcutsEnabled: function() {
      return this.$store.getters.getEnableKeyboardShortcuts
    },
```

File: src/renderer/components/ft-video-player/ft-video-player.js [hunk 6]
```diff
@@ -2087,8 +2090,8 @@ export default defineComponent({
       if ((event.ctrlKey || event.metaKey) && event.key.toLowerCase() === 'c') {
         return
       }
-
-      if (this.player !== null) {
+      // 'this.disableKeyboardShortcuts === false' address disabling keyboard shortcuts
+      if (this.player !== null && this.disableKeyboardShortcuts === false) {
```
Comment (PikachuEXE) [inline src/renderer/components/ft-video-player/ft-video-player.js:2094]:
```suggestion
      if (this.player !== null && this.keyboardShortcutsEnabled) {
```

File: src/renderer/components/player-settings/player-settings.js [hunk 7]
```diff
@@ -321,6 +325,7 @@ export default defineComponent({
       'updateScreenshotFilenamePattern',
       'parseScreenshotCustomFileName',
       'updateCommentAutoLoadEnabled',
+      'updateDisableKeyboardShortcuts'
```
Comment (PikachuEXE) [inline src/renderer/components/player-settings/player-settings.js:328]:
```suggestion
      'updateDisableKeyboardShortcuts',
```

File: src/renderer/components/player-settings/player-settings.vue [hunk 8]
```diff
@@ -87,6 +87,12 @@
           :default-value="enterFullscreenOnDisplayRotate"
           @change="updateEnterFullscreenOnDisplayRotate"
         />
+        <ft-toggle-switch
+          :label="$t('Settings.Player Settings.Disable Keyboard Shortcuts')"
```
Comment (PikachuEXE) [inline src/renderer/components/player-settings/player-settings.vue:91]:
```suggestion
          :label="$t('Settings.Player Settings.Enable Keyboard Shortcuts')"
```

File: src/renderer/store/modules/settings.js [hunk 9]
```diff
@@ -226,6 +226,7 @@ const state = {
   hideLabelsSideBar: false,
   hideChapters: false,
   showDistractionFreeTitles: false,
+  disableKeyboardShortcuts: false,
```
Comment (PikachuEXE) [inline src/renderer/store/modules/settings.js:229]:
```suggestion
  enableKeyboardShortcuts: true,
```

File: static/locales/en-US.yaml [hunk 10]
```diff
@@ -257,6 +257,7 @@ Settings:
     Turn on Subtitles by Default: Turn on Subtitles by Default
     Autoplay Videos: Autoplay Videos
     Proxy Videos Through Invidious: Proxy Videos Through Invidious
+    Disable Keyboard Shortcuts: Disable Keyboard Shortcuts
```
Comment (PikachuEXE) [inline static/locales/en-US.yaml:260]:
```suggestion
    Enable Keyboard Shortcuts: Enable Keyboard Shortcuts
```

File: src/renderer/components/player-settings/player-settings.vue [hunk 11]
```diff
@@ -87,6 +87,12 @@
...
+          :compact="true"
+          :default-value="disableKeyboardShortcuts"
```
Comment (PikachuEXE) [inline src/renderer/components/player-settings/player-settings.vue:93]:
```suggestion
          :default-value="enableKeyboardShortcuts"
```

File: src/renderer/components/player-settings/player-settings.vue [hunk 12]
```diff
@@ -87,6 +87,12 @@
...
+          @change="updateDisableKeyboardShortcuts"
```
Comment (PikachuEXE) [inline src/renderer/components/player-settings/player-settings.vue:94]:
```suggestion
          @change="updateEnableKeyboardShortcuts"
```

**Review by PikachuEXE — CHANGES_REQUESTED** [review]
Besides the series of distraction free settings
I see no reason to prefix a setting with a negative term (e.g. disable)

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@PikachuEXE wouldn't changing this to enable cause confusion for users?

**Comment by PikachuEXE** [comment]
There are many enabled by default settings
It's more confusing when it's "on" (except those "hide" prefix settings which is already another cluster) but it means disabled
2023-11-29 15_02_14-Window
2023-11-29 15_03_46-Window

And yes even the "smooth scrolling" one should be prefixed with "Enable" IMO (not saying it should be changed in this PR)
See how Chrome/Firefox display setting for "smooth scrolling"

Chrome
image

Firefox
image

**Comment by absidue** [comment]
I feel like in this context, using disable is okay, as I highly doubt most people will want to touch the setting, so making it clear that they are enabled by default seems like a good idea. We don't have any way of the user knowing what the default value is or whether they changed a setting from the default, which is how we end up with people turning on proxy videos through Invidious, complaining that stuff doesn't work, when the instance gets IP blocked and then blaming us for turning it on by default (which we don't).

Nobody has provided a good reason for why you would even want to disable a useful feature (the issue has no description and the person opening the pull request, could be the same person looking at the account name), that as far as I see it, has no downsides.

TL;DR This looks like a setting that only one person wants, so we should make it obvious to users that the default is enabled. Putting "Enable" in front of a setting that is enabled by default, makes it seem like an extra, that people might unknowingly disable, however in this case the setting exists to disable expected behaviour.

**Comment by PikachuEXE** [comment]
Fine make the label and UI prefixed with disable but make the setting prefixed with enable

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> Fine make the label and UI prefixed with disable but make the setting prefixed with enable

Is this the only blocker of the PR?

**Comment by PikachuEXE** [comment]
Yup
